pubsub/callback/callback.go

```diff
@@ -0,0 +1,41 @@
+package callback
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/micromdm/micromdm/pubsub"
+)
+
+type CallbackPublisher struct {
+	url  string
+	next pubsub.Publisher
+}
+
+func (p *CallbackPublisher) Publish(ctx context.Context, topic string, msg []byte) error {
+
+	// send to callback url
+	var buf bytes.Buffer
+	var event = struct {
+		Topic   string `json:"topic"`
+		Message []byte `json:"message"`
+	}{
+		Topic:   topic,
+		Message: msg,
```

Comment:
Given your requirements, my counter proposal would be to implement an HTTP middleware around the connect handler which forwards the request body from `/mdm/connect` to this callback url. It can default to just the raw body, but we can also add a flag for a json re-encode. 

IMO that would be the simplest solution that achieves all the desired goals with minimal disruption.